website/docs/api/top-level.mdx

```diff
@@ -340,6 +340,127 @@ use with the `manual=True` argument in `displacy.render`.
+| -------------- | ----------------------------------------------------------------------------------------------------------- |
+| `words`        | List of dictionaries describing a word token (see structure below). ~~List[Dict[str, Any]]~~                |
+| `arcs`         | List of dictionaries describing the relations between words (see structure below). ~~List[Dict[str, Any]]~~ |
+| `settings`     | Dependency Visualizer options (see [here](/api/top-level#displacy_options)). ~~Dict[str, Any]~~             |
+
+<Accordion title="Words data structure">
+
+| Dictionary Key | Description                              |
+| -------------- | ---------------------------------------- |
+| `text`         | Text content of the word. ~~str~~        |
+| `tag`          | Fine-grained part-of-speech. ~~str~~     |
+| `lemma`        | Base form of the word. ~~Optional[str]~~ |
+
+</Accordion>
+
+<Accordion title="Arcs data structure">
+
+| Dictionary Key | Description                                          |
+| -------------- | ---------------------------------------------------- |
+| `start`        | The index of the starting token. ~~int~~             |
+| `end`          | The index of the ending token. ~~int~~               |
+| `label`        | The type of dependency relation. ~~str~~             |
+| `dir`          | Direction of the relation (`left`, `right`). ~~str~~ |
+
+</Accordion>
+
+> #### ENT example data structure
+>
+> ```json
+>  {
+>    "text": "But Google is starting from behind.",
+>    "ents": [{"start": 4, "end": 10, "label": "ORG"}],
+>    "title": None
+>  }
+> ```
+
+#### Named Entity Recognition data structure {id="structure-ent"}
+
+| Dictionary Key | Description                                                                                 |
+| -------------- | ------------------------------------------------------------------------------------------- |
+| `text`         | String representation of the document text. ~~str~~                                         |
+| `ents`         | List of dictionaries describing entities (see structure below). ~~List[Dict[str, Any]]~~    |
+| `title`        | Title of the visualization. ~~str~~                                                         |
+| `settings`     | Entity Visualizer options (see [here](/api/top-level#displacy_options)). ~~Dict[str, Any]~~ |
+
+<Accordion title="Ents data structure">
+
+| Dictionary Key | Description                                         |
+| -------------- | --------------------------------------------------- |
+| `start`        | The index of the first token of the entity. ~~int~~ |
+| `end`          | The index of the last token of the entity. ~~int~~  |
+| `label`        | Label attached to the entity. ~~str~~               |
+| `kb_id`        | `KnowledgeBase` ID. ~~str~~                         |
+| `kb_url`       | `KnowledgeBase` URL. ~~str~~                        |
```

Comment:
I think these are both not required, so we might want to type as `Optional` to be consistent. However, this isn't exactly right because we're using `Optional` here to represent that the key doesn't need to be present, not that the type of the value is `Union[str, None]`.

Maybe we want another way to indicate the key is optional rather than the type. Like placing `(optional)` after the key name in the `Dictionary Key` column?

This is one area it would be good to have feedback on. I'm not sure if there's another place in the docs we have the same issue and what the practice is.  cc @svlandeg

Reply:
This is basically similar to kwd-args that are optional but not necessarily of type `Optional`. We denote those with a specific divider, cf https://spacy.io/api/language#init for example - so perhaps do something similar to that?

Reply:
This is great! Thanks, I'll try to add them with a divider

Reply:
I've added the `Optional` divider for the `KnowledgeBase` parameters and also for the `setting` parameter.